File: Homework/Manh/lesson-1/MoveZeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int num = 0;
        for(int i = 0; i < nums.length; i++){
            if(nums[i]!=0) {
                swap(i, num, nums);
                num++;
            }
        }
    }
    
    private void swap(int x, int y, int[] nums){
        if(x==y) return;
        nums[x] = nums[x] + nums[y];
        nums[y] = nums[x] - nums[y];
        nums[x] = nums[x] - nums[y];
    }
}

File: Homework/Khang_Nguyen/move-zeroes.js

/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
 const swap = (indexA, indexB, arr) => {
    const temp = arr[indexA]
    arr[indexA] = arr[indexB]
    arr[indexB] = temp
}
var moveZeroes = function(nums) {
  const len = nums.length
  let leftMostZeroIndex = -1;
  for (let i = 0; i < len; i++) {
      if (nums[i] === 0) {
          if (leftMostZeroIndex === -1) {
            leftMostZeroIndex = i            
          }
      } else if (leftMostZeroIndex !== -1) {
          swap(leftMostZeroIndex, i, nums)
          leftMostZeroIndex++
      }
  }
};